sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtilTest.java

```diff
@@ -58,4 +67,121 @@ void getOtlpProtocolDefault() {
+                ImmutableMap.of(
+                    "otel.exporter.otlp.endpoint", "http://localhost:4317/path",
+                    "otel.exporter.otlp.protocol", "http/protobuf")))
+        .doesNotThrowAnyException();
+  }
+
+  @Test
+  void configureOtlpExporterBuilder_InvalidEndpoints() {
+    assertThatThrownBy(
+            configureEndpoint(ImmutableMap.of("otel.exporter.otlp.endpoint", "/foo/bar")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must be a valid URL:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of("otel.exporter.otlp.endpoint", "file://localhost:4317")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint scheme must be http or https:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of("otel.exporter.otlp.endpoint", "http://localhost:4317?foo=bar")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a query string:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of("otel.exporter.otlp.endpoint", "http://localhost:4317#fragment")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a fragment:");
+    assertThatThrownBy(
+            configureEndpoint(
+                ImmutableMap.of(
+                    "otel.exporter.otlp.endpoint", "http://localhost:4317/path",
+                    "otel.exporter.otlp.protocol", "grpc")))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("OTLP endpoint must not have a path:");
+  }
+
+  private static ThrowingCallable configureEndpoint(Map<String, String> properties) {
+    return () ->
+        OtlpConfigUtil.configureOtlpExporterBuilder(
+            DATA_TYPE_TRACES,
+            DefaultConfigProperties.createForTest(properties),
+            value -> {},
+            (value1, value2) -> {},
+            value -> {},
+            value -> {},
+            value -> {});
+  }
+
+  @Test
+  void configureOtlpExporterBuilder_HttpProtobufEndpoint() {
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317"))
+        .isEqualTo("http://localhost:4317/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/"))
+        .isEqualTo("http://localhost:4317/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/foo"))
+        .isEqualTo("http://localhost:4317/foo/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/foo/"))
+        .isEqualTo("http://localhost:4317/foo/v1/traces");
+    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/v1/traces"))
+        .isEqualTo("http://localhost:4317/v1/traces/v1/traces");
```

Comment:
```suggestion
    assertThat(configureHttpProtobufEndpoint(DATA_TYPE_TRACES, "http://localhost:4317/v1/traces"))
        .isEqualTo("http://localhost:4317/v1/traces");
```

The spec mandates that signal-specific URLs are left alone and /v1/traces is *not* appended. What if you have an endpoint that needs an URL not ending in that? I don't think the OTLP spec mandates a particular URL path.

Reply:
> The per-signal endpoint configuration options take precedence and can be used to override this behavior (the URL is used as-is for them, without any modifications). 

and

> 1. For the per-signal variables (OTEL_EXPORTER_OTLP_<signal>_ENDPOINT), the URL MUST be used as-is without any modification. The only exception is that if an URL contains no path part, the root path / MUST be used (see Example 2).

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp (and table above)

Reply:
OK, maybe I misunderstand what the test does because the implementation at L56 in `sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java` looks like it's doing the right thing, not appending the signal path for the per-signal config.

Reply:
yes, it's quite hard to figure out what cases are being tested in these tests, at least from the github diff.

Reply:
Maybe a better function name could already help?

Reply:
yes, maybe `configureHttpProtobufEndpoint` could be renamed to `configureNonSignalSpecificEndpoint` or something to make it super clear in the `OtlpConfigUtilTest`

Reply:
> OK, maybe I misunderstand what the test does because the implementation at L56 in sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java looks like it's doing the right thing, not appending the signal path for the per-signal config.

Yes these test cases are specifically testing the behavior of the signal agnostic `otel.exporter.otlp.endpoint` property. 

I can try to improve the names of the function(s) to improve clarity.